srv/salt/_modules/cephinspector.py

```diff
@@ -6,17 +6,41 @@
 from subprocess import Popen, PIPE
 import json
 
+def _get_disk_id(partition):
+    """
+    Return the disk id of a partition/device, or an empty string if not available.
+    """
+    disk_id_cmd = Popen("find -L /dev/disk/by-id -samefile " + partition + " \( -name ata* -o -name nvme* \)", stdout=PIPE, stderr=PIPE, shell=True)
+    out, err = disk_id_cmd.communicate()
+
+    # We should only ever have one entry that we return.
+    return out.rstrip()
+
 def _append_to_ceph_disk(ceph_disks, partition, journal_dev):
     """
     Populate ceph_disks dictionary with data and journal partitions.
     """
+
+    # We don't care about the trailing number on journal_dev.
+    journal_dev = journal_dev.rstrip("1234567890")
+    # For nvme journal, strip the trailing 'p' as well.
+    if "nvme" in journal_dev:
+        journal_dev = journal_dev[:-1]
```

Comment:
As far as I understand, our (DeepSea's) storage role yml files operate on base device for data and journal paritions.